src/repositories/user.repository.ts

```diff
@@ -30,4 +41,41 @@ export class UserRepository extends DefaultCrudRepository<
       async () => orderRepository,
     );
   }
+
+  async login(credentials: Credentials) {
+    // Validate Email
+    if (!isemail.validate(credentials.email)) {
+      throw new HttpErrors.UnprocessableEntity('invalid email');
+    }
+
+    // Validate Password Length
+    if (credentials.password.length < 8) {
+      throw new HttpErrors.UnprocessableEntity(
+        'password must be minimum 8 characters',
+      );
+    }
+
+    // Check if user exists
+    try {
+      const foundUser = await this.findOne({
+        where: {email: credentials.email, password: credentials.password},
+      });
+
+      let token = undefined;
+      if (foundUser) {
+        const currentUser = _.pick(toJSON(foundUser), ['id', 'email']);
+        // Generate user token using JWT
+        token = await signAsync(currentUser, 'secretforjwt', {
+          expiresIn: 300,
+        });
+      } else {
+        return Promise.reject(
+          new HttpErrors.Unauthorized('Wrong credentials!'),
+        );
+      }
+      return Promise.resolve(token);
```

Comment:
No need to wrap the value in `Promise.resolve`, async functions always return promises. Just `return token`.